lib/run-tests/worker.js

```diff
@@ -1,10 +1,39 @@
 const path = require("path");
+const crypto = require("crypto");
+const lmdb = require("lmdb");
+
+const dependencies = require("eshost/lib/dependencies");
 
 const BabelAgent = require("./babel-agent");
 const transpile = require("./transpile");
 
 const TEST_TIMEOUT = 60 * 1000; // 1 minute
 
+function sha1(str) {
+  return crypto.createHash("sha1").update(str, "utf8").digest("hex");
+}
+
+function buildCacheHash(test) {
+  let { phase = "", type = "" } = test.attrs.negative || {};
+  if ((phase === "early" || phase === "parse") && type === "SyntaxError") {
+    return sha1(test.contents);
+  }
+  const sources = dependencies
+    .getDependencies(test.file)
+    .sort()
+    .map(function (file) {
+      if (file === path.basename(test.file)) {
+        return;
+      }
+      return dependencies.rawSource.get(path.basename(file));
+    })
+    .filter(Boolean);
+
+  sources.push(test.contents);
+
+  return sha1(sources.join("\n"));
+}
```

Comment:
Now it uses the same method as eshost to get dependencies, which is safe enough to merge. :)